### ISSUE TITLE
TCPConvergenceLayer binds only once

### ISSUE THREAD

**Issue by dklimkin** [ibrdtn/ibrdtn#252]
$ dtnd ...
$ netstat -anp | grep LISTEN
tcp        0      0 127.0.0.1:4550          0.0.0.0:*               LISTEN      1279/dtnd
tcp        0      0 172.17.0.2:4556         0.0.0.0:*               LISTEN      1279/dtnd
$ ifconfig eth0 down
$ ifconfig eth0 up
$ netstat -anp | grep LISTEN
tcp        0      0 127.0.0.1:4550          0.0.0.0:*               LISTEN      1279/dtnd
$

This breaks communication for configurations where links go up and down regularly, which DTN should tolerate.

**Comment by dklimkin** [comment]
LinkManager sends two events: ACTION_LINK_UP, ACTION_LINK_RUNNING but never ACTION_ADDRESS_ADDED. 

TCPConvergenceLayer::eventNotify handles neither of former.

**Comment by morgenroth** [comment]
This had worked in the past. Have you tested this with the latest release?

**Comment by dklimkin** [comment]
You mean not git HEAD but a release? This might have worked with Posix manager, it's failing for me with libnl.

**Comment by morgenroth** [comment]
Yes, I mean not the HEAD. It had worked with libnl before and the question is why it stopped working.

**Comment by dklimkin** [comment]
I'll check tomorrow, but I don't think it did work. To clarify, this happens if the interface is fully put down. Bringing up / shutting down aliases works as expected. 

With `ifconfig eth0 up` netlink just doesn't generate RTM_NEWADDR.

**Comment by morgenroth** [comment]
Usually, network managers set an interface 'up' first and then assign an IP address. Isn't this the case in your test?

**Comment by dklimkin** [comment]
This interface has a static IP address. But it can go up and down.

**Comment by morgenroth** [comment]
Even with ifup/ifdown the address is removed in added again after the interfaces up. So, please and more details on your problem. I think it's more artificial in real world.

**Comment by dklimkin** [comment]
It's very real on a stripped down system that doesn't run a network manager. Running /etc/init.d/scripts also doesn't produce this event.

I can generate change addr event by `ifconfig eth0 some_other_ip` but it's an ugly workaround. And the patch handles link up/down well. Any reason you would like to handle LINK_DOWN but not LINK_UP?

The original comment has specific steps to reproduce, could you please try them? I believe the same effect would be on physical link lost, when no dhcp is used.

**Comment by shortstheory** [comment]
I have a similar problem which relates to this. We are trying to use IBR-DTN using Linux nodes which connect to each other using WiFi direct using wpa_supplicant. On forming a P2P group, the device creates a new interface such as p2p-wlp3s0-X where the X changes on every new group created (i.e. when a device goes out of range and tries connecting to another device using WiFi direct). This is a problem for us as we have to restart the DTND client each time the interface changes. Is there any way to change the interface DTND listens on while DTND is running?

**Comment by morgenroth** [comment]
The API is made to handle P2P connections with changing interfaces. Connect via TCP to port 4550 and say... 

```
protocol p2p_extension wifi
interface up p2p-wlp3s0-X
```

to add a new interface and

```
interface down p2p-wlp3s0-X
```

to remove it again.
